### PR TITLE
Make quote characters configurable

### PR DESCRIPTION
**Pull request by dmnd**
Moves the quote characters to config. Motivated by #14, but does not change the
default to include backticks.

Test Plan:

Ran specs. Let me know if you think we need more specs. I only added one.

### CODE REVIEW

**Comment by dmnd** [comment]
rebased this on top of latest master (to fix deprecations)

File: lib/toggle-quotes.coffee [hunk 1]
```diff
@@ -26,25 +29,30 @@ toggleQuoteAtPosition = (editor, position) ->
...
+  newText = prefix + nextQuoteCharacter + newText[(1+prefix.length)...-1] + nextQuoteCharacter
 
   editor.setTextInBufferRange(range, newText)
 
-getOppositeQuote = (quoteCharacter) ->
-  if quoteCharacter is '"'
-    "'"
-  else
-    '"'
+getNextQuote = (c, cs) ->
```
Comment (kevinsawicki) [inline lib/toggle-quotes.coffee]: Mind using full variable names here? Like `quoteCharacter` and `quoteCharacters` perhaps?

File: lib/toggle-quotes.coffee [hunk 2]
```diff
@@ -13,7 +14,9 @@ toggleQuoteAtPosition = (editor, position) ->
     # This is useful for languages where changing the quotes makes the range
     # invalid and so toggling again should properly restore the valid quotes
     if range = editor.displayBuffer.bufferRangeForScopeAtPosition('.invalid.illegal', position)
-      return unless /^(".*"|'.*')$/.test(editor.getTextInBufferRange(range))
+      inner = quoteChars.split('').map((c) -> "#{c}.*#{c}").join('|')
+      regex = new RegExp("^(#{inner})$", 'g')
```
Comment (kevinsawicki) [inline lib/toggle-quotes.coffee]: CoffeeScript has a nice `///` option for interpolated regexes, so this could be:

```coffee
///^(#{inner})$///g
```

**Comment by kevinsawicki** [comment]
Sorry for the delay, this seems good to me, just a few minor comments to address, thanks.

File: lib/toggle-quotes.coffee (hunk 2 above)
Comment (dmnd) [inline lib/toggle-quotes.coffee]: TIL, thanks!

File: lib/toggle-quotes.coffee (hunk 1 above)
Comment (dmnd) [inline lib/toggle-quotes.coffee]: Done, but fwiw I find `c` and `cs` much easier to distinguish from one another than `quoteCharacter` and `quoteCharacters`

**Comment by dmnd** [comment]
No worries about delay. Thanks for looking at this!

File: lib/toggle-quotes.coffee (hunk 1 above)
Comment (kevinsawicki) [inline lib/toggle-quotes.coffee]: Yeah, fair point, maybe `quoteCharacter` and `allQuoteCharacters` then?

Comment (dmnd) [inline lib/toggle-quotes.coffee]: :+1: done!

**Comment by kevinsawicki** [comment]
Great, thanks for this!

**Comment by kevinsawicki** [comment]
Published in 0.10

**Comment by dmnd** [comment]
Thank you :smile:

**Comment by mnquintana** [comment]
@dmnd @kevinsawicki Should this setting be visible in the Packages details view for toggle-quotes? Because I'm not seeing anything (even after restarting Atom and reinstalling toggle-quotes). :confused:

**Comment by dmnd** [comment]
@mnquintana This is a bundled package so I think you have to wait until the version in core gets bumped.

**Comment by mnquintana** [comment]
@dmnd It isn't yet: https://github.com/atom/atom/blob/master/package.json#L118-L119 and #15

I just downloaded the latest release (0.10.0), but for some reason the setting isn't rendering in the view.

**Comment by dmnd** [comment]
@mnquintana Oops, my mistake. I just reinstalled and I see this:

image

So I don't know why that doesn't work for you. I'm on Atom 0.176.0, what are you running?

**Comment by mnquintana** [comment]
@dmnd Same here. This is what I'm getting:
screenshot 2015-01-27 23 09 32

**Comment by dmnd** [comment]
Hmm. I'm guessing your config is probably messed up? E.g. why do you have 2 identical keybinds there?

**Comment by mnquintana** [comment]
Ah, found it – [save-session](https://github.com/mpeterson2/save-session) was caching an old version of settings-view for some reason, so it wasn't showing the new settings.

**Pull request merged**